### DISCUSSION TITLE
How to have dynamic path segments like /foo-{x}-bar-{y} in Next.js?

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#38194] [Help] [answered]
Is it possible to create dynamic path segments with multiple variables, like?

```
/{type}-is-{feature}-unless-{status}
/foo-{x}-bar-{y}
```

Basically, I want to plug in an arbitrary number of variables into the path, maybe with multiple path segments where each segment has multiple variables in it. Is this possible, or if not, planned? If not, why not?

**Comment by icyJoseph** [comment, marked as answer]
Mmm, not sure what you are up to, but you could:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return [
      {
        source: "/:type(\\w+)-is-:feature(\\w+)",
        destination: "/multiple",
      },
    ];
  },
};

module.exports = nextConfig;
```

When in `pages/multiple.tsx` I can do:

> browser url: http://localhost:3001/foo-is-bar

```tsx
import type { GetServerSidePropsContext, NextPage } from "next";
import Link from "next/link";
import { useRouter } from "next/router";

const Multiple: NextPage = () => {
  console.log(useRouter().query); // {type: 'foo', feature: 'bar'}
  return (
    <div>
      Multiple <Link href="/blog">blog</Link>
    </div>
  );
};

export const getServerSideProps = (ctx: GetServerSidePropsContext) => {
  console.log(ctx.query); // { type: 'foo', feature: 'bar' }
  return { props: {} };
};

export default Multiple;
```

**Reply by icyJoseph** [reply]
If you want more complex stuff, perhaps look at https://github.com/pillarjs/path-to-regexp

**Comment by GabenGar** [comment]
You shouldn't use paths like this because it's going to be a nightmare to work with. Either use /segment/path/s or just good old search params for that. Otherwise create a variable per path segment and then parse them as strings however you like.

**Comment by lancejpollard** [comment]
https://webmasters.stackexchange.com/a/139776/82253
